KOGITO-2237: Stunner - Domain Model Improvements

romartin opened the pull request:
See https://issues.redhat.com/browse/KOGITO-2237

romartin commented:
Jenkins execute full downstream build

hasys commented on kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/definition/adapter/binding/ClientBindingUtils.java:
```diff
@@ -85,30 +110,7 @@
-            }
-        }
-        return null;
+    private static boolean isNotEmpty(final String s) {
```
You can use `org.kie.workbench.common.stunner.core.util.StringUtils#nonEmpty`

hasys commented on kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-core-common/src/main/java/org/kie/workbench/common/stunner/core/definition/clone/DeepCloneProcess.java:
```diff
@@ -50,29 +52,24 @@ public DeepCloneProcess(final FactoryManager factoryManager,
-                .forEach(entry -> {
-                    Object value = adapterManager.forProperty().getValue(entry.getKey());
-                    adapterManager.forProperty().setValue(entry.getValue(),
-                                                          cloneValue(value));
```
`cloneValue` private method is not used any more and can be deleted.

hasys commented on kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-api/kie-wb-common-stunner-core-api/src/main/java/org/kie/workbench/common/stunner/core/definition/adapter/binding/DefinitionAdapterBindings.java:
```diff
@@ -0,0 +1,238 @@
+            metaTypeFieldIndexes[HEIGHT] = index;
+        }
+
+        public void setRadiustIndex(int index) {
```
Typo in method name

romartin commented:
Hey @hasys - I've jsut rebased and fixed your review comments, thanks!

Hey @karreiro - added u also as a review for the DMN part, is it ok? This PR is about internal updates only, so everything should work as before

Thanks guys!

romartin commented:
Jenkins execute full downstream build

karreiro commented:
Jenkins execute full downstream build

karreiro commented:
Jenkins execute full downstream build.

hasys commented:
Jenkins execute full downstream build

romartin commented:
Rebased guys, let's see the build now
Thanks!

romartin commented:
Jenkins execute full downstream build

hasys commented:
Jenkins execute compile downstream build

karreiro commented:
Jenkins execute compile downstream build.

romartin commented:
Jenkins execute compile downstream build

romartin commented:
rebased again @karreiro , let's see if we can get some success build ... :/

romartin commented:
Jenkins execute compile downstream build

romartin commented:
Jenkins execute full downstream build

karreiro commented:
Jenkins execute full downstream build

romartin commented:
ooops @karreiro , honestly I'm just testing the `dmn-webapp-kogito-runtime`... and it's working perfectly! So sorry if any inconveniences, lemme test other webapps and let you know in a while.
Thanks for pointing this out!!!

romartin commented:
oh thanks a lot @karreiro , great catch! I've been testing and the webapps which were using backend services were broken. I've added the fix. I've tested the DMN showcases and are working properly.

Thanks a lot again for this and sorry for the inconveniences. Fix pushed and ready for review again!! :+1:

romartin commented:
Jenkins execute compile downstream build

romartin commented:
Jenkins execute full downstream build

romartin commented:
Jenkins execute compile downstream build

romartin commented:
Jenkins execute compile downstream build

romartin commented:
Hey @karreiro 

Can you please provide some feedback on this? I would like trying to merge before going to vacation. Thanks!

romartin commented:
thanks @karreiro , good catch again! :+1: 
This looks not a regular "resize" operation, so not sure about what could be wrong yet, just lemme take a quick look on it... thx again! Will also try to do also more tests specially on DMN side, trying not to spend your time on iteration reviews.. sorry for this.

romartin commented:
Jenkins execute compile downstream build

romartin commented:
Hey @karreiro 

Thanks a lot for your help on this!! I've properly fixed the issue you mentioned, also did a deep look on code (based on this fix) and everything looks good. Also tested both standalone and kogito showcases for DMN and working properly, at least doing some tests.

Sooooo, please this should be ready for review again, sorry for bothering!!!

Thanks again!!

romartin commented:
Hey @karreiro 

Thanks for the feedback again! I've properly fixed the issue in the way that no other webapps (like the drools-wb or jbpm-wb) will be affected. I've properly tested and working fine.

So I've  rebased, fixed and squashed all commits, everything is ready for review again. 

Thanks again for the collaboration here!

romartin commented:
Jenkins execute fdb

romartin commented:
Jenkins execute cdb

romartin commented:
Hey @karreiro 

Thanks for the help! Anyway I couldn't reproduce the issue you mention in the `drools-wb`. I can properly open diagrams, create, save, etc. Any concrete step to reproduce? Are you using an empty project? Clean app?

I've rebased and force pushed, let's see if that fixes the issue you mention.

Thx!

romartin commented:
Jenkins execute fdb

romartin commented:
Jenkins execute cdb

mbiarnes commented:
FDB test comment

mbiarnes commented:
FDB test comment

romartin commented:
Merging. I've tested the full bulid locally and it works.
